Build failure: megasync

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *megasync*

### Build log

```
mega/src/gfx/freeimage.cpp:316:44: error: invalid conversion from 'const AVCodec*' to 'AVCodec*' [-fpermissive]
  316 |     AVCodec* decoder = avcodec_find_decoder(codecId);
      |                        ~~~~~~~~~~~~~~~~~~~~^~~~~~~~~
      |                                            |
      |                                            const AVCodec*
```

### Additional context

f4af3b77fd86f33239fe5770f9eaba3ecef0de11 builds fine

### Notify maintainers

@michojel

### Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.2.7-zen1, NixOS, 23.05 (Stoat), 23.05.20230330.e3652e0`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos, nixos-22.11-small-22.11-small"`
 - nixpkgs: `/run/current-system/nixpkgs`

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -5267,7 +5267,7 @@ with pkgs;
   medusa = callPackage ../tools/security/medusa { };
 
   megasync = libsForQt5.callPackage ../applications/misc/megasync {
-    ffmpeg = ffmpeg-full;
+    ffmpeg = ffmpeg_4;
   };
 
   megacmd = callPackage ../applications/misc/megacmd { };

